src/classes/CON_Phone.cls

```diff
@@ -183,6 +183,16 @@ public with sharing class CON_Phone {
                 contact.OtherPhone = contact.Phone;          
         }
     }
+
+    /*******************************************************************************************************
+    * @description Returns a map with Preferred Phone picklist entry's label as key and API as value
+    * @retun Map<String, String> A key value pair of Preferred Phone picklist entrys's label and API respectively
+    */
+    private static Map<String, String> createPreferredLabelToAPIMap() {             
```

Comment:
I'm sorry for holding up the QA on this and adding a last minute suggestion (& also sorry for going AFK immediately after our meeting today!)

When I had mentioned caching performance enhancements here, I meant that it would be nice to only generate the map once. This will be especially useful if we have a large number of contacts that we're iterating over at once.

I won't hold up the review for this, but a future enhancement here could be something like:
```
private static Map<String, String> labelToAPIMap;
private static Map<String, String> getPreferredLabelToAPIMap() {
    if (labelToAPIMap == null) {
        // code goes here
        labelToAPIMap = UTIL_Describe...
    }
    return labelToAPIMap;
}
```

In this example, the map is only generated once and subsequent calls to `getPreferredLabelToAPIMap` return the stored value.

Reply:
Thanks for the clarification @robertbwatson. I made the change, can you take another quick look? Thx.

Reply:
Perfect @skosgi1 -- thanks!